server/lib/src/server/mod.rs

```diff
@@ -1500,8 +1513,33 @@ impl<'a> QueryServerWriteTransaction<'a> {
     pub(crate) fn reload_domain_info(&mut self) -> Result<(), OperationError> {
         let domain_name = self.get_db_domain_name()?;
         let display_name = self.get_db_domain_display_name()?;
+        let domain_ldap_allow_unix_pw_bind = match self.get_domain_ldap_allow_unix_pw_bind() {
+            Ok(v) => {
+                v
+            },
+            _ => {
+                admin_warn!("Defaulting ldap_allow_unix_pw_bind to true");
+
+                let modl = ModifyList::new_purge_and_set(
```

Comment:
Don't do this internal_modify here - you are *reading* the state of the boolean, not updating it. The reason for this is to consider "on upgrade defaults".

If you *write* to the database, we can't tell if the write was from us or the user, so we have to remain stuck on the default.

But if we allow the bool to be absent from the entry, and we just "unwrap_or(true)", then we can later change the "absent" state from true -> false, and every user gets the default changed. Had you written that to the db we can't do that!

Reply:
Ahhh, had my suspicions that the writing defaults to database was causing some messups...this makes sense!